### PR TITLE
Type decl kind annotation

### PR DESCRIPTION
**Pull request by ersran9**
Optional kind annotations for data / newtype declarations. Can be extended for class and instance decls.

### CODE REVIEW

**Comment by ersran9** [comment]
@googlebot I signed it!

**Comment by judah** [comment]
Thank you for this change!  I thought it over and realized it would be more consistent to make all of the functions like `class'`, `data'`, etc. take `HsTyVarBndr'` directly instead of strings, the same way that `forall'` already does.  That would be a breaking change, but I think a minor one: we'd just replace `data' "Maybe" ["a"] ...` with `data' "Maybe" [bvar "a"] ...`

I've pushed that change here (building on top of yours):
https://github.com/google/ghc-source-gen/compare/kinded-var

Please let us know whether that would satisfy your use-case; if so, I can merge my change.

File: tests/pprint_test.hs [hunk 1]
```diff
@@ -279,6 +279,29 @@ declsTest dflags = testGroup "Decls"
         [ "pattern F a b = G b a"
             :~ patSynBind "F" ["a", "b"] $ conP "G" [bvar "b", bvar "a"]
         ]
+    , test "dataDecl"
```
Comment (judah) [inline tests/pprint_test.hs]: Thank you for adding these tests!

**Comment by ersran9** [comment]
This works for us. Thanks!

**Comment by ersran9** [comment]
I'll close this PR. I have one more change for promoted tuple types (https://github.com/byteally/ghc-source-gen/commit/94deed1175a9ebd38b518969faece2f5b02436a0) . Once your change is merged in, I'll rebase and send you a new PR. Thanks for making this library!

**Comment by judah** [comment]
Thanks, I've merged #61 with the changes I mentioned.

**Pull request closed without merging**